ctru-rs/examples/camera-image.rs

```diff
@@ -0,0 +1,214 @@
+use ctru::console::Console;
+use ctru::gfx::{RawFrameBuffer, Screen, Side};
+use ctru::services::cam::{
+    Cam, CamContext, CamOutputFormat, CamPort, CamSelect, CamShutterSoundType, CamSize,
+};
+use ctru::services::hid::KeyPad;
+use ctru::services::{Apt, Hid};
+use ctru::Gfx;
+
+const WIDTH: i16 = 400;
+const HEIGHT: i16 = 240;
+const SCREEN_SIZE: u32 = 192000; // WIDTH * HEIGHT * 2;
+const BUF_SIZE: usize = 384000; // SCREEN_SIZE * 2;
+
+const WAIT_TIMEOUT: i64 = 300000000;
+
+fn main() {
+    ctru::init();
+
+    let apt = Apt::init().expect("Failed to initialize Apt service.");
+    let hid = Hid::init().expect("Failed to initialize Hid service.");
+    let gfx = Gfx::init().expect("Failed to initialize GFX service.");
+
+    gfx.top_screen.borrow_mut().set_double_buffering(true);
+    gfx.bottom_screen.borrow_mut().set_double_buffering(false);
+
+    let _console = Console::init(gfx.bottom_screen.borrow_mut());
+
+    let mut key_down;
+    let mut key_held;
+
+    println!("Initializing camera");
+
+    let mut cam = Cam::init().expect("Failed to initialize CAM service.");
+
+    cam.set_size(
+        CamSelect::SELECT_OUT1_OUT2,
+        CamSize::SIZE_CTR_TOP_LCD,
+        CamContext::CONTEXT_A,
+    )
+    .expect("Failed to set camera size");
+    cam.set_output_format(
+        CamSelect::SELECT_OUT1_OUT2,
+        CamOutputFormat::OUTPUT_RGB_565,
+        CamContext::CONTEXT_A,
+    )
+    .expect("Failed to set camera output format");
+
+    cam.set_noise_filter(CamSelect::SELECT_OUT1_OUT2, true)
+        .expect("Failed to enable noise filter");
+    cam.set_auto_exposure(CamSelect::SELECT_OUT1_OUT2, true)
+        .expect("Failed to enable auto exposure");
+    cam.set_auto_white_balance(CamSelect::SELECT_OUT1_OUT2, true)
+        .expect("Failed to enable auto white balance");
+
+    cam.set_trimming(CamPort::PORT_CAM1, false)
+        .expect("Failed to disable trimming for Cam Port 1");
+    cam.set_trimming(CamPort::PORT_CAM2, false)
+        .expect("Failed to disable trimming for Cam Port 2");
+
+    let mut buf = vec![0u8; BUF_SIZE];
+
+    gfx.flush_buffers();
+    gfx.swap_buffers();
+    gfx.wait_for_vblank();
+
+    let mut held_r = false;
+
+    println!("\nPress R to take a new picture");
+    println!("Press Start to exit to Homebrew Launcher");
+
+    gfx.top_screen.borrow_mut().set_3d_enabled(false);
```

Comment:
What's the purpose of this? It's off by default.

Reply:
I was testing stuff with 3D and forgot to remove it.

Decided against adding 3D because the example created a raw pointer to the slider value rather than using a lib function to retrieve it. Removing this line now.